pytest often fails with KeyError: 'Contents'

# Problems

pytest often fails with `KeyError: 'Contents'`.  

# Possible causes

This repository currently will test with 3 of python versions when pushed to GitHub or Pull Request was created.
The unit tests run on each testing environments at the same time, so if one of that environments accessed files, it might happen that the files do not exist .

# Error messages

```shell

Run pip install -e .
Obtaining file:///home/runner/work/photo-backup/photo-backup
Installing collected packages: photo-backup
  Running setup.py develop for photo-backup
Successfully installed photo-backup-0.1.0
============================= test session starts ==============================
platform linux -- Python 3.9.5, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/runner/work/photo-backup/photo-backup
collected 28 items

test_exif.py .................                                           [ 60%]
test_photo.py .........F                                                 [ 96%]
test_upload.py F                                                         [100%]

=================================== FAILURES ===================================
_________________________ TestPhotosUpload.test_upload _________________________

self = <test_photo.TestPhotosUpload object at 0x7fdc12938070>

    def test_upload(self):
        photos = Photos(S3Uploader(consts.S3_BUCKET))
    
        photo1 = Photo("data/images/IMG_9235.jpeg")
        photos.append(photo1)
    
        photo2 = Photo("data/images/SOVK6553.jpeg")
        photos.append(photo2)
    
        photos.upload()
    
        s3 = boto3.client("s3")
        response = s3.list_objects_v2(Bucket=consts.S3_BUCKET, Prefix="2020")
        uploaded_filename = [
>           content["Key"] for content in response["Contents"]
        ]
E       KeyError: 'Contents'

test_photo.py:105: KeyError
_______________________ TestS3UploaderUpload.test_upload _______________________

self = <test_upload.TestS3UploaderUpload object at 0x7fdc1019aca0>

    def test_upload(self):
        expected = "2020/08/20200813140708.jpeg"
    
        uploader = S3Uploader(consts.S3_BUCKET)
        uploader.upload("data/images/IMG_9235.jpeg", expected)
    
        s3 = boto3.client("s3")
        response = s3.list_objects_v2(
            Bucket=consts.S3_BUCKET, Prefix="2020/08"
        )
>       uploaded_filename = response["Contents"][0]["Key"]
E       KeyError: 'Contents'

test_upload.py:18: KeyError
=========================== short test summary info ============================
FAILED test_photo.py::TestPhotosUpload::test_upload - KeyError: 'Contents'
FAILED test_upload.py::TestS3UploaderUpload::test_upload - KeyError: 'Contents'
========================= 2 failed, 26 passed in 6.64s =========================
Error: Process completed with exit code 1.
```

--- a/requirements.txt
+++ b/requirements.txt
@@ -27,6 +27,7 @@ pycodestyle==2.7.0
 pyflakes==2.3.1
 pyparsing==2.4.7
 pytest==6.1.2
+pytest-mock==3.6.1
 python-dateutil==2.8.1
 PyYAML==5.4.1
 regex==2021.4.4

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,0 +1,8 @@
+import uuid
+
+import pytest
+
+
+@pytest.fixture(scope="function")
+def test_folder():
+    return uuid.uuid4()

--- a/tests/test_photo.py
+++ b/tests/test_photo.py
@@ -88,28 +88,32 @@ class TestPhotosAppend:
 
 
 class TestPhotosUpload:
-    def test_upload(self):
+    def test_upload(self, test_folder, mocker):
         photos = Photos(S3Uploader(consts.S3_BUCKET))
 
+        expected = [
+            f"test/{test_folder}/2020/08/20200813140708.jpeg",
+            f"test/{test_folder}/2020/10/20201012131028.jpeg",
+        ]
+
         photo1 = Photo("data/images/IMG_9235.jpeg")
+        mocker.patch.object(photo1, "create_dst_path", lambda: expected[0])
         photos.append(photo1)
 
         photo2 = Photo("data/images/SOVK6553.jpeg")
+        mocker.patch.object(photo2, "create_dst_path", lambda: expected[1])
         photos.append(photo2)
 
         photos.upload()
 
         s3 = boto3.client("s3")
-        response = s3.list_objects_v2(Bucket=consts.S3_BUCKET, Prefix="2020")
+        response = s3.list_objects_v2(
+            Bucket=consts.S3_BUCKET, Prefix=f"test/{test_folder}/2020"
+        )
         uploaded_filename = [
             content["Key"] for content in response["Contents"]
         ]
         uploaded_filename.sort()
-
-        expected = [
-            "2020/08/20200813140708.jpeg",
-            "2020/10/20201012131028.jpeg",
-        ]
 
         assert uploaded_filename == expected
 
@@ -119,7 +123,7 @@ class TestPhotosUpload:
                 "s3",
                 "rm",
                 "--recursive",
-                f"s3://{consts.S3_BUCKET}/2020/",
+                f"s3://{consts.S3_BUCKET}/test/{test_folder}/2020/",
             ]
         )
         response.check_returncode()

--- a/tests/test_upload.py
+++ b/tests/test_upload.py
@@ -5,15 +5,15 @@ from photo_backup.upload import S3Uploader
 
 
 class TestS3UploaderUpload:
-    def test_upload(self):
-        expected = "2020/08/20200813140708.jpeg"
+    def test_upload(self, test_folder):
+        expected = f"test/{test_folder}/2020/08/20200813140708.jpeg"
 
         uploader = S3Uploader(consts.S3_BUCKET)
         uploader.upload("data/images/IMG_9235.jpeg", expected)
 
         s3 = boto3.client("s3")
         response = s3.list_objects_v2(
-            Bucket=consts.S3_BUCKET, Prefix="2020/08"
+            Bucket=consts.S3_BUCKET, Prefix=f"test/{test_folder}/2020/08"
         )
         uploaded_filename = response["Contents"][0]["Key"]
 

